src/docfx/publish/PublishModelBuilder.cs

```diff
@@ -133,6 +124,17 @@ public bool TryAdd(Document file, PublishItem item)
             var fileManifests = _publishItems.ToDictionary(item => item.Key, item => item.Value);
 
             return (model, fileManifests);
+
+            void HandleFileWithError(Document file)
```

Comment:
prefer member function over inner function here

Reply:
what's the reason?

Reply:
It just makes it harder to reason about correctness. Scanning local functions for captured variables is no fun.
https://github.com/dotnet/csharplang/issues/1277 https://github.com/dotnet/csharplang/issues/1565

Reply:
hmm, if we want to avoid the inner function, can we add it to our develop guideline?  seems we are use to use inner function, right?

Reply:
I think we should not forbid inner function here. For instance, in the case of recursion, we use inner function to avoid passing along too many inputs.